### PR TITLE
Various fixes/improvements for domain selector

### PR DESCRIPTION
**Pull request by Polymorphe57**
Task ID: 3324924

### CODE REVIEW

File: addons/web/static/src/search/search_model.js [hunk 1]
```diff
@@ -206,7 +206,7 @@ export function getDomainTreeDescription(
         const values = Array.isArray(value) ? value : [value];
         const formatValue = (value) => {
             if (typeof value === "string") {
-                return `"${value}"`;
+                return `${value}`;
```
Comment (mcm-odoo) [inline addons/web/static/src/search/search_model.js:267]: `return value;`

**Review by mcm-odoo — COMMENTED** [review]
These commits could be squashed together to prevent commit without message:
- d4ba1e6b140153f9de80d0d82fd13e949167cd75 -> dc04a37cba7bade00bf5ba2244db62c283cd4e01
- a235aef9bee235e3715775b4c4647f977c2560cf -> fb58a8c38b6fee3f4f5feafaec83bfe783153c29
- 6d95d58eb2e8a1dbb8d89f5b6aa0c90f0a26efea -> 1497c72d52deed9db08cf4ae793ebcec18924ef1

File: addons/web/static/src/core/domain_selector/domain_selector.xml [hunk 2]
```diff
@@ -13,15 +13,14 @@
...
-                    Match records with the following rule:
-                    <t t-call="web._DomainSelector.branch.children" />
-                </t>
                 <t t-else="">
                     <span>Match records with </span>
                     <t t-call="web._DomainSelector.branch.connector" />
                     <span> of the following rules:</span>
                     <t t-call="web._DomainSelector.branch.children" />
+                    <t t-if="!props.readonly">
+                        <a href="#" class="pt-1 ms-1" style="display: block;" role="button" t-on-click="() => this.insertRootLeaf(node)">New Rule</a>
```
Comment (Polymorphe57) [inline addons/web/static/src/core/domain_selector/domain_selector.xml:22]: d-block (in class)

**Comment by mcm-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**